File: Try_GameEngine5/src/com/example/try_gameengine/utils/ISpriteDetectAreaListener.java
package com.example.try_gameengine.utils;

/**
 * 
 * @author irons
 *
 */
public interface ISpriteDetectAreaListener {
//	public boolean detectedResult(DetectArea handlerDetectArea, IDetectAreaRequest requestDetectArea, boolean isDetected);
	/**
	 * @param handlerDetectArea
	 * @param requestDetectArea
	 */
	public void didDetected(DetectArea handlerDetectArea, IDetectAreaRequest requestDetectArea);
	/**
	 * @param handlerDetectArea
	 * @param requestDetectArea
	 * @param isDetected
	 * @return
	 */
	public boolean stopDoSuccessorDetected(DetectArea handlerDetectArea, IDetectAreaRequest requestDetectArea, boolean isDetected);
}

File: Try_GameEngine5/src/com/example/try_gameengine/utils/ISpriteDetectAreaListener.kt
package com.example.try_gameengine.utils

/**
 * 
 * @author irons
 // */
interface ISpriteDetectAreaListener {
    //	public boolean detectedResult(DetectArea handlerDetectArea, IDetectAreaRequest requestDetectArea, boolean isDetected);
    /**
     * @param handlerDetectArea
     * @param requestDetectArea
     // */
    fun didDetected(handlerDetectArea: DetectArea?, requestDetectArea: IDetectAreaRequest?)

    /**
     * @param handlerDetectArea
     * @param requestDetectArea
     * @param isDetected
     * @return
     // */
    fun stopDoSuccessorDetected(
        handlerDetectArea: DetectArea?,
        requestDetectArea: IDetectAreaRequest?,
        isDetected: Boolean
    ): Boolean
}
